aiorospy/src/aiorospy/topic.py

```diff
@@ -1,15 +1,28 @@
+import asyncio
 import janus
 import rospy
 
 
 class AsyncSubscriber:
 
-    def __init__(self, name, data_class):
-        self._queue = janus.Queue()
-        self._subscriber = rospy.Subscriber(name, data_class, lambda msg: self._queue.sync_q.put(msg))
-
+    def __init__(self, name, data_class, queue_size=None, loop=None):
+        self._loop = loop if loop is not None else asyncio.get_running_loop()
+        self._queue = janus.Queue(maxsize=queue_size, loop=loop)
+        self._subscriber = rospy.Subscriber(name, data_class, queue_size=queue_size, callback=self._callback)
         self.unregister = self._subscriber.unregister
 
+    def _callback(self, msg):
```

Comment:
Is this implementation of queue_size used or tested anywhere?

Reply:
Nope, you said you would add tests for topics/services ;) I can write one if you like

Reply:
I was more curious why this ended up in this PR. Doesn't really matter though.  Thanks for the tests.